src/main/java/frc/robot/utils/logging/LoggingCommand.java

```diff
@@ -0,0 +1,113 @@
+package frc.robot.utils.logging;
+
+import edu.wpi.first.util.datalog.BooleanLogEntry;
+import edu.wpi.first.wpilibj.DataLogManager;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.Subsystem;
+
+import java.util.Set;
+
+public abstract class LoggingCommand extends Command {
+    private String namePrefix;
+    private String commandName;
+    private final Command underlying;
+
+    private String fullyQualifiedName;
+    // Lazy initialized log entry to make sure we don't create it until it is needed (and command is scheduled)
+    // otherwise we get an empty line in the log visualization tool
+    private BooleanLogEntry logEntry;
+
+    public LoggingCommand(String namePrefix, String commandName, Command underlying) {
+        this.namePrefix = namePrefix;
+        this.commandName = commandName;
+        this.underlying = underlying;
+        resetLoggingName(namePrefix, commandName);
+    }
+
+    @Override
+    public void initialize() {
+        getLoggingEntry().append(true);
```

Comment:
need to add `if (Constants.LOGGING_ENABLED)`

Reply:
done